inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,22 +26,22 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
-    file_name: str
-    sources: list[str] = field(default_factory=lambda: [])
+    source_list_file_name: str
+    sources: List[str] = field(default_factory=list)
+    gpg_key_uri: Optional[str] = field(default=None)
+    gpg_key_name: Optional[str] = field(default=None)
 
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationRemoveSourceParameters:
-    gpg_key_name: str
-    file_name: str
+    source_list_file_name: str
+    gpg_key_name: Optional[str] = field(default=None)
 
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationUpdateSourceParameters:
-    file_name: str
-    sources: list[str] = field(default_factory=lambda: [])
+    source_list_file_name: str
+    sources: List[str] = field(default_factory=list)
```

Comment:
**Enhancement**
**Suggestion:** To prevent any accidental mutation of the sources list after the instance is created, consider using a tuple instead of a list for the sources field in ApplicationUpdateSourceParameters.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    @dataclass(kw_only=True, frozen=True)
    class ApplicationUpdateSourceParameters:
        sources: Tuple[str, ...] = field(default_factory=tuple)
```